### DISCUSSION TITLE
Struggling with "cloning": fine-tuning vs. YourTTS, embeddings, oh my?!

### DISCUSSION THREAD

**Discussion by thoraxe** [coqui-ai/TTS#2429] [General Q&A]
I'm probably about six to eight hours in, which I realize is not a lot, but I am really just going in circles at this point.

I am trying to experiment with voice cloning. I have an RTX3080 with Cuda enabled on Windows for training.

In my first experiment, I have about an hour of audio that is not in dataset form. I have pulled a bunch of clips so far and built an LJSpeech-type dataset, but it's only about two minutes of audio (1:48, to be accurate) over 20 clips right now. I know that I likely need to build a larger dataset here.

The goal is to come up with "models" (if that is the right term) representing four different sports commentators to then use in "real-time" (Python flask request/response API) to generate sports commentary.

I have a lot of questions, and I cannot seem to find clear answers, hence the circles.

* It's not clear if I should be fine-tuning or just using the YourTTS cloning
* It's not clear, due to a lack of documentation, how much audio input YourTTS needs. 60 seconds of input didn't seem to produce any meaningful approximation of the source voice (heavy British-English accent) and was, in some regards, worse than only 5 seconds.
* It looks like I can fine-tune YourTTS, but it's not clear whether I need to do the entire VCTK training on top of my subset or if there's a way to start with the existing published model. `train_yourtts.py` does indicate transfer learning is possible from the original model, and I'm downloading VCTK right now (it's part of that script) to attempt to do just that.
* I realize that much more audio is better, but something like https://github.com/CorentinJ/Real-Time-Voice-Cloning shows you can clone reasonably well with only a few seconds of audio -- so what am I doing wrong with YourTTS which is referred to by CorentinJ?

This is probably mostly questions for @Edresson - I am happy to add/write additional documentation or blogs once I get this all figured out.

Update as of 3/21/2023:
I now have two bugs open. One is for computing embeddings, and the other is for trying to train ("fine tune") YourTTS:
https://github.com/coqui-ai/TTS/issues/2444
https://github.com/coqui-ai/TTS/issues/2428

**Comment by thelabcat** [comment]
I'm also having trouble with voice cloning, but mostly in that accents seem to be completely ignored.

**Comment by p0p4k** [comment]
What does this mean?

> It's not clear if I should be fine-tuning or just using the YourTTS cloning

**Reply by p0p4k** [reply]
If you mean train yourtts from scratch, then of course the voice wont be learnt. YourTTS is a model that has few-shot voice cloning capabilities, meaning you train the model on a big dataset and then with very little samples, a fine-tuned yourtts model is capable of generating the small dataset voices, essentially "voice cloning".
